rust/chains/abacus-ethereum/src/trait_builder.rs

```diff
@@ -0,0 +1,59 @@
+use std::sync::Arc;
+
+use ethers::prelude::*;
+
+use abacus_core::{ContractLocator, Signers};
+
+use crate::Connection;
+
+async fn make_signing_provider<M: Middleware>(
+    provider: M,
+    signer: Signers,
+) -> Result<SignerMiddleware<NonceManagerMiddleware<M>, Signers>, M::Error> {
+    let provider_chain_id = provider.get_chainid().await?;
+    let signer = ethers::signers::Signer::with_chain_id(signer, provider_chain_id.as_u64());
+
+    let address = ethers::prelude::Signer::address(&signer);
+    let provider = NonceManagerMiddleware::new(provider, address);
+
+    let signing_provider = SignerMiddleware::new(provider, signer);
+    Ok(signing_provider)
+}
+
+pub(crate) trait MakeableWithProvider {
+    type Output;
+
+    fn make<M: Middleware + 'static>(self, provider: M, locator: &ContractLocator) -> Self::Output;
```

Comment:
Love the un-macro quest that you are on. Is `make` the best name for this though? Maybe indicate what it makes? i.e. `createContractInstance`? (i'm known to be a terrible namer)

Reply:
I mean, the name of a crate-private trait function that is only used in specific places isn't that big of a deal. But if you want `make_contract` would also make sense and be slightly more verbose, however the main purpose of the `make` function here should be conveyed by the trait itself since it is a one-function trait.

Reply:
Yeah I guess the comment extrapolates to the name of the trait itself as well, `MakeableWithProvider` sounds quite abstract